### PR TITLE
Update FileWriter buffer size in storage from int64_t to size_t.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Part of https://github.com/Azure/azure-sdk-for-cpp/issues/2226

Mainly modifies FileWriter and call sites.

Todo: Changelog

cc @Jinming-Hu

### CODE REVIEW

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-common/src/file_io.cpp [hunk 1]
```diff
@@ -155,15 +155,13 @@ namespace Azure { namespace Storage { namespace _internal {
 
   FileWriter::~FileWriter() { close(m_handle); }
 
-  void FileWriter::Write(const uint8_t* buffer, int64_t length, int64_t offset)
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/src/file_io.cpp:158]: @Jinming-Hu can the offset be made `size_t` as well?

Also, can you take over this branch/PR and make the changes to the callsite as part of https://github.com/Azure/azure-sdk-for-cpp/issues/2313?

For example, it requires changes here:
https://github.com/Azure/azure-sdk-for-cpp/blob/e5763720d4a08a90b2e5e63525957e5051387430/sdk/storage/azure-storage-files-shares/src/share_file_client.cpp#L802-L815

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/src/file_io.cpp:158]: > can the offset be made size_t as well?

Yes, I think so.

> Also, can you take over this branch/PR and make the changes to the callsite as part of #2313?

Do you want to me append commits to this PR or open a new PR?

Comment (ahsonkhan) [reply sdk/storage/azure-storage-common/src/file_io.cpp:158]: Whatever's easier for you. Create a new PR, and I will close this one. :)

**Comment by Jinming-Hu** [comment]
I changed the type for `length` parameter in `FileWriter::Write()` from `size_t` to `length` in this PR https://github.com/Azure/azure-sdk-for-cpp/pull/2320

**Pull request closed without merging**